Packages/com.unity.inputsystem/InputSystem/Plugins/Switch/NPad.cs

```diff
@@ -1,776 +0,0 @@
-    {
-        public static FourCC Type => new FourCC("SXSP");
-        internal const int kSize = InputDeviceCommand.kBaseCommandSize;
-
-        [FieldOffset(0)]
-        public InputDeviceCommand baseCommand;
-
-        public FourCC typeStatic
-        {
-            get { return Type; }
-        }
-
-        public static NpadDeviceIOCTLStopSixAxisSensor Create()
-        {
-            return new NpadDeviceIOCTLStopSixAxisSensor
-            {
-                baseCommand = new InputDeviceCommand(Type, kSize),
-            };
-        }
-    }
-    /// <summary>
-    /// Switch output report sent as command to backend.
-    /// </summary>
-    // IMPORTANT: Struct must match the NpadDeviceIOCTLOutputReport in native
-    [StructLayout(LayoutKind.Explicit, Size = kSize)]
-    internal struct NPadDeviceIOCTLOutputCommand : IInputDeviceCommandInfo
-    {
-        public static FourCC Type { get { return new FourCC('N', 'P', 'G', 'O'); } }
-
-        internal const int kSize = InputDeviceCommand.kBaseCommandSize + 20;
-        public const float DefaultFrequencyLow = 160.0f;
-        public const float DefaultFrequencyHigh = 320.0f;
-
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1028:EnumStorageShouldBeInt32", Justification = "Need to match native struct data size")]
-        public enum NPadRumblePostion : byte
-        {
-            Left = 0x02,
-            Right = 0x04,
-            All = 0xFF,
-            None = 0x00,
-        }
-
-        [FieldOffset(0)] public InputDeviceCommand baseCommand;
-
-        [FieldOffset(InputDeviceCommand.kBaseCommandSize + 0)]
-        public byte positions;
-        [FieldOffset(InputDeviceCommand.kBaseCommandSize + 4)]
-        public float amplitudeLow;
-        [FieldOffset(InputDeviceCommand.kBaseCommandSize + 8)]
-        public float frequencyLow;
-        [FieldOffset(InputDeviceCommand.kBaseCommandSize + 12)]
-        public float amplitudeHigh;
-        [FieldOffset(InputDeviceCommand.kBaseCommandSize + 16)]
-        public float frequencyHigh;
-
-        public FourCC typeStatic
-        {
-            get { return Type; }
-        }
-
-        public static NPadDeviceIOCTLOutputCommand Create()
-        {
-            return new NPadDeviceIOCTLOutputCommand()
-            {
-                baseCommand = new InputDeviceCommand(Type, kSize),
-                positions = (byte)NPadRumblePostion.None,
-                amplitudeLow = 0,
-                frequencyLow = DefaultFrequencyLow,
-                amplitudeHigh = 0,
-                frequencyHigh = DefaultFrequencyHigh
-            };
-        }
-    }
-#endif
-
-#if UNITY_EDITOR || UNITY_STANDALONE_WIN || UNITY_STANDALONE_OSX || UNITY_WSA
-    /// <summary>
-    /// Structure of HID input reports for Switch Pro controllers.
-    /// </summary>
-    [StructLayout(LayoutKind.Explicit, Size = 20)]
-    internal struct SwitchProControllerHIDInputState : IInputStateTypeInfo
```

Comment:
We want to keep this, and move it to a separate file - this is the  Switch controller HID implementation used on Desktops!